posthog/clickhouse/test/test_cluster.py

```diff
@@ -0,0 +1,65 @@
+import uuid
+from collections.abc import Iterator
+
+import pytest
+from clickhouse_driver import Client
+
+from posthog.clickhouse.cluster import ClickhouseCluster, MutationRunner, get_cluster
+from posthog.models.event.sql import EVENTS_DATA_TABLE
+
+
+@pytest.fixture
+def cluster(django_db_setup) -> Iterator[ClickhouseCluster]:
+    yield get_cluster()
+
+
+def test_mutations(cluster: ClickhouseCluster) -> None:
+    table = EVENTS_DATA_TABLE()
+    count = 100
+
+    # make sure there is some data to play with first
+    def populate_random_data(client: Client) -> None:
+        client.execute(f"INSERT INTO {table} SELECT * FROM generateRandom() LIMIT {count}")
+
+    cluster.map_one_host_per_shard(populate_random_data).result()
+
+    # construct the runner
+    sentinel_uuid = uuid.uuid1()  # unique to this test run to ensure we have a clean slate
+    runner = MutationRunner(table, "UPDATE person_id = %(uuid)s WHERE 1 = 1", {"uuid": sentinel_uuid})
+
+    # nothing should be running yet
+    existing_mutations = cluster.map_all_hosts(runner.find).result()
+    assert all(mutation is None for mutation in existing_mutations.values())
+
+    # start all mutations
+    shard_mutations = cluster.map_one_host_per_shard(runner.enqueue).result()
+    assert len(shard_mutations) > 0
+
+    # check results
+    def get_person_ids(client: Client) -> list[tuple[uuid.UUID, int]]:
+        return client.execute(f"SELECT person_id, count() FROM {table} GROUP BY ALL ORDER BY ALL")
+
+    for host_info, mutation in shard_mutations.items():
+        assert host_info.shard_num is not None
```

Comment:
This is to resolve this type error:

```
Error: posthog/clickhouse/test/test_cluster.py:44: error: Argument 1 to "map_all_hosts_in_shard" of "ClickhouseCluster" has incompatible type "int | None"; expected "int"  [arg-type]
```

Asserting like this is a bit of a hack: I'll refactor this at some point (later - not here), and make it so that there is a shard object returned (instead of a host) and make that able to be like `shard.map_hosts(fn)` and `shard.any_host(fn)`, a bit like it's own little constrained `ClickhouseCluster` version. That feels a lot more natural to work with than having to pass the shard back in each time.